tooling-server-configuration/src/main/java/org/jboss/eap/qe/microprofile/tooling/server/configuration/ConfigUtils.java

```diff
@@ -0,0 +1,127 @@
+package org.jboss.eap.qe.microprofile.tooling.server.configuration;
+
+import org.jboss.arquillian.config.descriptor.api.ArquillianDescriptor;
+import org.jboss.arquillian.config.descriptor.api.ContainerDef;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.event.suite.BeforeSuite;
+
+import java.util.Optional;
+
+/**
+ * Provides access to configuration properties
+ */
+public class ConfigUtils {
+
+    public static final String DEFAULT_CONTAINER_NAME = "jboss";
+    public static final String DEFAULT_MANAGEMENT_ADDRESS_VALUE = "127.0.0.1";
+    public static final String DEFAULT_MANAGEMENT_PORT_VALUE = "9990";
+    public static final String ARQ_MANAGEMENT_ADDRESS_PROPERTY_NAME = "managementAddress";
+    public static final String ARQ_MANAGEMENT_PORT_PROPERTY_NAME = "managementPort";
+
+    private ConfigUtils() {}
+
+    /**
+     * This property is initialized during Before* phase by ArquillianConfiguration extension
+     */
+    private static ArquillianDescriptor arquillianDescriptor;
```

Comment:
I suggest to move this `ArquillianDescriptor` obtaining logic to separate class - something like `ArquillianDescriptorWrapper` and then force user to inject the instance to this "utility" class to prevent inconsistent states and make code more readable. This would include creating a proper constructor and not making all related methods static.

Reply:
There were probably some little misunderstanding. By "user" I mean a programmer which will be using this API. By "inject" I mean "dependency injection" [1] - just pass the `ArquillianDescriptor` as a parameter to this utility class constructor.

[1] https://en.wikipedia.org/wiki/Dependency_injection

Reply:
Hi @honza-kasik I got both the meanings and context, thanks... 
I was just wondering about why injecting in constructor and not, let's say, the field.

Reply:
Because you are calling static methods here which are relying on a fact that static field is not null. Without someone not following whole process how it is initialized they don't know how can they rely on this not being null. This is why I suggest separating field value population and utility class and forcing user to do a bit coding more so the whole thing is more readable after all.

Reply:
I've removed the `static`s, too :)

Reply:
Done. 
Given the current PR state + changes you suggested, `ManagmentClientProvider` is still used through its static methods, which - in turn - use `ArquillianContainerProperties` (formerly known as `ConfigUtils`), which is now non-static - i.e. descriptor passed to constructor.
In order to let the programmer supply the static`ArquillianDescriptorWrapper#getArquillianDescriptor()` value to an `ArquillianContainerProperties` instance constructor, a static field  in test class is initialized and then passed to `ManagmentClientProvider` factory methods (e.g. `ManagmentClientProvider#onlineStandalone`).
Please let me know whether you believe this should be adjusted further.